> (Apologies, got cut-off in the middle of review) @HeetVekariya looks like you might need to install the documentation dependencies, something like `pip install -e ".[doc]"` in the `package/` directory should install everything you need to build docs.

- I have successfully run ```pip install -e ".[doc]"```
- To run ```make html```, first i have commented [this](https://github.com/MDAnalysis/mdanalysis/blob/37d8fee8e54af08aafe1ad1349811a714bbd87e2/package/doc/sphinx/source/conf.py#L193) line, as i don't have any static folder.
- Then i run ```make html``` which gives this error

```
reading sources... [ 99%] documentation_pages/visualization/streamlines_3D
reading sources... [ 99%] documentation_pages/visualization_modules
reading sources... [100%] index

sphinx-sitemap: No pages generated for sitemap.xml

Traceback (most recent call last):
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/cmd/build.py", line 298, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/application.py", line 355, in build
    self.builder.build_update()
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 293, in build_update
    self.build(to_build,
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/builders/__init__.py", line 312, in build
    with logging.pending_warnings():
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/contextlib.py", line 142, in __exit__
    next(self.gen)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/util/logging.py", line 222, in pending_warnings
    memhandler.flushTo(logger)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/util/logging.py", line 187, in flushTo
    logger.handle(record)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/logging/__init__.py", line 1634, in handle
    self.callHandlers(record)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/logging/__init__.py", line 1696, in callHandlers
    hdlr.handle(record)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/logging/__init__.py", line 964, in handle
    rv = self.filter(record)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/logging/__init__.py", line 821, in filter
    result = f.filter(record)
  File "/home/heet/anaconda3/envs/mdanalysis-dev/lib/python3.10/site-packages/sphinx/util/logging.py", line 427, in filter
    raise exc
sphinx.errors.SphinxWarning: /home/heet/Work/Development/mdanalysis/package/MDAnalysis/coordinates/FHIAIMS.py:docstring of MDAnalysis.coordinates.base.ProtoReader.timeseries:8:Explicit markup ends without a blank line; unexpected unindent.

Warning, treated as error:
/home/heet/Work/Development/mdanalysis/package/MDAnalysis/coordinates/FHIAIMS.py:docstring of MDAnalysis.coordinates.base.ProtoReader.timeseries:8:Explicit markup ends without a blank line; unexpected unindent.
make: *** [Makefile:42: html] Error 2
```

--- a/package/AUTHORS
+++ b/package/AUTHORS
@@ -222,6 +222,7 @@ Chronological list of authors
   - Shubham Kumar
   - Zaheer Timol
   - Geongi Moon
+  - Heet Vekariya
 
 External code
 -------------

--- a/package/MDAnalysis/exceptions.py
+++ b/package/MDAnalysis/exceptions.py
@@ -41,9 +41,8 @@ class NoDataError(ValueError, AttributeError):
       (e.g., bonds, charges) is not present in the data.
 
     * Raised when data is missing in an analysis class because the `run()`
-      method has not been called. Examples include polymer analysis, PSA
-      (Path Similarity Analysis), dielectric analysis, and hydrogen bond
-      analysis.
+      method has not been called, e.g.
+      :class:`~MDAnalysis.analysis.polymer.PersistenceLength`.
 
     * Raised when timestep data is missing, such as positions, velocities,
       or forces.

